### PR TITLE
Local binary install 

### PR DESCRIPTION
**Pull request by potemkin-git**
https://github.com/rgsystemes/kb/issues/7118

### CODE REVIEW

**Comment by potemkin-git** [comment]
@sandvige 

The "zip download + extract" intends to pass Choco CI which runs install without (token) parameter. Purpose is to pass tests with a successful installation-ish.
Only `Install-ChocolateyZipPackage` seems to allow it without actually attempting an install (https://docs.chocolatey.org/en-us/create/functions/)
This command uses download only.
Its alternative `Get-ChocolateyUnzip` may trigger a successful install too (I didn't test but guessing "no") but involves embedding a 50Mb nupkg instead of 6Mb without it. As it's more like a bad use case imho, I chose to leave it outside.

Last: url https://github.com/rgsystemes/rgsupv-chocolatey/raw/main/rgsupv-win.zip is a guess I've anticipated but not tested yet 😇

File: tools/chocolateyuninstall.ps1 [hunk 1]
```diff
@@ -1,5 +1,11 @@
-Uninstall-ChocolateyZipPackage 'rgsupervision' 'rgsupv-win.zip'
-
 If (Test-Path -Path "C:\Program Files (x86)\RG-Supervision\RG_Supervision.exe") {
-	& "C:\Program Files (x86)\RG-Supervision\RG_Supervision.exe" --action uninstall-console
+	$packageArgs = @{
+		packageName    = "rgsupervision"
+		fileType       = "exe"
+		file           = "C:\Program Files (x86)\RG-Supervision\RG_Supervision.exe"
+		silentArgs     = "--action uninstall-console"
+		validExitCodes = @(0, 3010, 1641)
```
Comment (sandvige) [inline tools/chocolateyuninstall.ps1]: What are theses codes?

File: tools/chocolateyinstall.ps1 [hunk 2]
```diff
@@ -1,40 +1,44 @@
...
+  $packageArgs = @{
+    packageName    = $packageName
+    fileType       = "exe"
+    file           = $fileLocation
+    file64         = $fileLocation
+    checksum       = $checkSum
+    checksum64     = $checkSum
+    checksumType   = $checkSumType
+    silentArgs     = $silentArgs
+    validExitCodes = @(0, 3010, 1641)
```
Comment (sandvige) [inline tools/chocolateyinstall.ps1]: Same here?

File: tools/chocolateyuninstall.ps1 (hunk 1 above)
Comment (potemkin-git) [reply tools/chocolateyuninstall.ps1]: Hmmm probably kept it from chocolatey doc but after quick check it looks like MSI packages error codes
Don't know RG_Supervision.exe possible exit codes, should I keep 0 and 1 ?

File: tools/chocolateyinstall.ps1 (hunk 2 above)
Comment (potemkin-git) [reply tools/chocolateyinstall.ps1]: Hmmm probably kept it from chocolatey doc but after quick check it looks like MSI packages error codes
Don't know RG-Setup.exe possible exit codes, should I keep 0 and 1 ?

File: tools/chocolateyuninstall.ps1 (hunk 1 above)
Comment (potemkin-git) [reply tools/chocolateyuninstall.ps1]: Well well, no need to know all possible codes, only 0 is success I guess so > removed
image

File: tools/chocolateyinstall.ps1 (hunk 2 above)
Comment (potemkin-git) [reply tools/chocolateyinstall.ps1]: image

Comment (sandvige) [reply tools/chocolateyinstall.ps1]: validExitCode should cover "valid exit code for success", or any return status?

File: tools/chocolateyuninstall.ps1 (hunk 1 above)
Comment (sandvige) [reply tools/chocolateyuninstall.ps1]: Here is the code describing it: https://github.com/rgsystemes/rgsupv-agent/blob/master/agent/main.cpp#L330-L338, which refers to CLIController, which only use EXIT_SUCCESS (0) or EXIT_FAILURE (1).

**Review by sandvige — APPROVED** [review]

**Review by sandvige — APPROVED** [review]

**Pull request merged**